src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1189,19 +1190,30 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 Icon icon = chooser.getIcon(file);
                 setIcon(icon);
 
-            } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+            } else if (value instanceof Long len) {
                 if (listViewWindowsStyle) {
-                    text = MessageFormat.format(kiloByteString, len + 1);
-                } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    if (len == 0) {
+                        text = MessageFormat.format(kiloByteString, len);
+                    } else {
+                        len /= 1000L;
+                        text = MessageFormat.format(kiloByteString, len + 1);
+                    }
+                } else if (len < 1000L) {
+                    text = MessageFormat.format(kiloByteString, (len==0 ? 0L : 1L));
                 } else {
-                    len /= 1024L;
-                    if (len < 1024L) {
-                        text = MessageFormat.format(megaByteString, len);
+                    double kbVal = formatToDoubleValue(len);
+                    len = (long)kbVal;
```

Comment:
This code still looks sus to me - there is no need to assign len (it goes out of scope pretty quickly) and formatToDoubleValue() should just format and return a formatted String.

Reply:
As the returned value has been used in comparison with 1000.0 , so returning a double value from formatToDoubleValue().

Reply:
> Looks like it matches on ubuntu.
> 
> Do we care about other linuxes / desktop environments / file browsers?

I checked in Oracle Linux and it is showing file size similar to native.